examples/pytorch/ogb/ogbn-mag/hetero_rgcn.py

```diff
@@ -383,27 +394,49 @@ def main(args):
 
     for run in range(args.runs):
 
-        embed_layer.reset_parameters()
-        model.reset_parameters()
+        try:
+            embed_layer.reset_parameters()
+            model.reset_parameters()
+        except:
+            # old pytorch version doesn't support reset_parameters() API
+            pass
 
         # optimizer
         all_params = itertools.chain(
             model.parameters(), embed_layer.parameters()
         )
         optimizer = th.optim.Adam(all_params, lr=0.01)
 
-        logger = train(
-            g,
-            model,
-            embed_layer,
-            optimizer,
-            train_loader,
-            split_idx,
-            labels,
-            logger,
-            device,
-            run,
-        )
+        if args.num_workers != 0 and device == "cpu" and is_support_affinity(v_t):
+            expected_max = int(psutil.cpu_count(logical=False)/2)
+            if args.num_workers > expected_max:
+                print(f"[ERROR] You specified num_workers are larger than physical cores, please set any number less or equal to {expected_max}", file=sys.stderr)
+            with train_loader.enable_cpu_affinity():
```

Comment:
@chang-l , I cleaned args in `enable_cpu_affinity()` and add some error msg in case num_workers is mis-configured.